crypto/crypto_dh.c

```diff
@@ -127,11 +127,15 @@ blinded_modexp(uint8_t r[CRYPTO_DH_PUBLEN], BIGNUM * a,
 		goto err8;
 	}
 	rlen = BN_num_bytes(r1);
+	if (rlen < 0) {
+		warn0("Unexpected error in OpenSSL");
+		goto err8;
+	}
 	if (rlen > CRYPTO_DH_PUBLEN) {
 		warn0("Exponent result too large!");
 		goto err8;
 	}
-	memset(r, 0, CRYPTO_DH_PUBLEN - rlen);
+	memset(r, 0, CRYPTO_DH_PUBLEN - (size_t) rlen);
 	BN_bn2bin(r1, r + CRYPTO_DH_PUBLEN - rlen);
```

Comment:
I looked this up previously:
```
void *memset(void *s, int c, size_t n);
int BN_bn2bin(const BIGNUM *a, unsigned char *to);
```
Evidently clang is happy to mix signed and unsigned ints when it comes to pointer math?